### PR TITLE
refactoring-some-code

### PR DESCRIPTION
**Pull request by vldmrklnk**

### CODE REVIEW

File: InternshipBE/BL/DTOs/BestContactTimeDTO.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace BL.DTOs
+{
+	public class BestContactTimeDTO
+	{
+		private DateTime startTime;
```
Comment (romanstat) [inline InternshipBE/BL/DTOs/BestContactTimeDTO.cs]: _startTime

File: InternshipBE/BL/DTOs/BestContactTimeDTO.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		{
+			get
+			{
+				return endTime;
+			}
+			set
+			{
+				endTime = DateTime.Parse(value.ToString("G"));
+			}
+		}
```
Comment (romanstat) [inline InternshipBE/BL/DTOs/BestContactTimeDTO.cs]: use lambda

File: InternshipBE/BL/DTOs/EventDTO.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		{
+			get
+			{
+				return endTime;
+			}
+			set
+			{
+				endTime = DateTime.Parse(value.ToString("G"));
+			}
+		}
```
Comment (romanstat) [inline InternshipBE/BL/DTOs/EventDTO.cs]: same

File: InternshipBE/DAL/Migrations/20211114122733_sdfsfsfsdg.cs [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 namespace DAL.Migrations
 {
-    public partial class InitialDb : Migration
+    public partial class sdfsfsfsdg : Migration
```
Comment (romanstat) [inline InternshipBE/DAL/Migrations/20211114122733_sdfsfsfsdg.cs:6]: naming?

File: InternshipBE/DAL/Repositories/UserRepository.cs [hunk 5]
```diff
@@ -9,42 +9,42 @@
...
-        public async Task<List<User>> GetSpecificUsersByInternshipIdAsync(int id, RoleType? roleType = null)
-        {
-            var internship = await _context.Internships.AsNoTracking()
-                                                       .Include(x => x.Users.Where(u => u.RoleType == roleType || roleType == null))
-                                                       .FirstOrDefaultAsync(x => x.Id == id);
+		public async Task<List<User>> GetSpecificUsersByInternshipIdAsync(int id, RoleType? roleType = null)
+		{
+			var internship = await _context.Internships.AsNoTracking()
+													   .Include(x => x.Users.Where(u => u.RoleType == roleType || roleType == null))
+													   .FirstOrDefaultAsync(x => x.Id == id);
```
Comment (romanstat) [inline InternshipBE/DAL/Repositories/UserRepository.cs]: is it okey?

File: InternshipBE/WebApi/Controllers/GoogleCalendarController.cs [hunk 6]
```diff
@@ -19,7 +19,7 @@ public GoogleCalendarController(IGoogleCalendarService googleCalendarService)
 
 		[HttpPost("setEventToGoogleCalendar")]
 		[Authorize]
-		public async Task<IActionResult> SetEventToGoogleCalendar([FromBody] EventDTO model)
+		public async Task<IActionResult> SetEventToGoogleCalendarAsync([FromBody] EventDTO model)
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/GoogleCalendarController.cs]: remove Async in controllers

**Review by romanstat — APPROVED** [review]

File: InternshipBE/DAL/Repositories/BestContactTimeRepository.cs [hunk 7]
```diff
@@ -16,14 +16,13 @@ public BestContactTimeRepository(ApplicationDbContext context) : base(context)
...
-			return await _context.BestContactTimes.Include(x => x.User).ToListAsync();
+			return await _context.BestContactTimes.Include(x => x.User).Where(x => x.UserId == userId).ToListAsync();
 		}
 
 		public async Task<BestContactTime> GetByTimeIntervalAsync(BestContactTime contactTime)
 		{
-			return await _context.BestContactTimes.Where(x => x.StartTime.ToString("G") == contactTime.StartTime.ToString("G"))
-				.Where(x => x.EndTime.ToString("G") == contactTime.EndTime.ToString("G"))
-				.FirstAsync(x => x.UserId == contactTime.UserId);
+			return await _context.BestContactTimes.Where(x => x.StartTime == contactTime.StartTime && x.EndTime == contactTime.EndTime)
```
Comment (Amarozka123) [inline InternshipBE/DAL/Repositories/BestContactTimeRepository.cs]: Looks like it should be Where(x => x.StartTime >= contactTime.StartTime && x.EndTime <= contactTime.EndTime)

**Review by Xodoc — APPROVED** [review]

**Review by Evgeniy-Sinicin — APPROVED** [review]

**Pull request merged**